docs/it-manual/api/program-applications.md

```diff
@@ -1,182 +1,781 @@
 
-**Example export**
+##### Repeated entity `object`s
+Each repeated entity is an object that follows the same semantics as the `application` object as a whole. It has an `entity_name` property, as well as a property for each repeated question.
 
+###### `entity_name`
+- **Property**: `entity_name`
+- **JSON Type**: `string`
+- **Format**: A Unicode `string` of any characters.
+- **Description**: The name the applicant provided for each repeated entity. Used on each repeated screen in the repeated question title, such as "What is `$household_member`'s phone number?"
+
+An enumerator question, with two entities and a repeated `household_member_phone_number` phone question and a repeated `household_member_job` text question, looks like
+
+{% code %}
+```json
+"household_members": {
+  "question_type": "ENUMERATOR",
+  "entities": [
+    {
+      "entity_name": "Taylor",
+      "household_member_job": {
+        "question_type": "TEXT",
+        "text": "Pop artist"
+      },
+      "household_member_phone_number": {
+        "question_type": "PHONE",
+        "phone_number": "+16156667777"
+      }
+    },
+    {
+      "entity_name": "Sza",
+      "household_member_job": {
+        "question_type": "TEXT",
+        "text": "R&B artist"
+      },
+      "household_member_phone_number": {
+        "question_type": "PHONE",
+        "phone_number": "+18623334444"
+      }
+    }
+  ]
+},
 ```
+{% endcode %}
+
+### File upload questions
+File upload questions are not currently available in the API. See [GitHub Issue #5025](https://github.com/civiform/civiform/issues/5025) for progress on this feature.
+
+### ID questions
+`"question_type": "ID"`
+
+In addition to the metadata field, ID questions have the following property:
+
+#### `id`
+- **Property**: `id`
+- **JSON Type**: `string` or `null`
+- **Format**: A Unicode string consisting of only digits (`^[0-9]*$`). If min or max lengths are specified on the question they are also enforced.  `null` if unanswered.\
+_Note_: Answers from applications from before a min or max constraint is added to a question will not follow that constraint and may be of any length.
+- **Description**: The applicant's answer to the ID question.
+
+An ID question looks like
+
+{% code %}
+```json
+"drivers_license_number" : {
+  "question_type" : "ID",
+  "id" : "011235813"
+},
+```
+{% endcode %}
+
+### Multi-select questions
+`"question_type": "MULTI_SELECT"`
+
+Multi-select questions allow applicants to select multiple options from a list.
+
+In addition to the metadata field, multi-select questions have the following property:
+
+#### `selections`
+- **Property**: `selections`
+- **JSON Type**: `[string]`
+- **Format**: An array of Unicode `string`s, which are the admin IDs of the selected options. An empty array `[]` if no options were selected or the question was unanswered. With the exception of legacy questions, admin IDs can only contain lowercase letters, numbers, underscores, and dashes.\
```

Comment:
For the spots that you mention `legacy questions` it might be better to says something akin to "questions created before DATE". It would be more clear what this pertains to and prevent confusion down the road if we have other things we end up referring to as legacy.